azure-pipelines.yml

```diff
@@ -0,0 +1,59 @@
+jobs:
+
+- job:
+  strategy:
+    matrix:
+      Mojave-Xcode-10.2.1:
+        IMAGE_POOL: 'macOS-10.14'
+        XCODE_VERSION: '10.2.1'
+      Mojave-Xcode-11.0:
+        IMAGE_POOL: 'macOS-10.14'
+        XCODE_VERSION: '11'
+  pool:
+    vmImage: $(IMAGE_POOL)
+  
+  steps:
+
+  - script: sudo xcode-select --switch "/Applications/Xcode_$(XCODE_VERSION).app/Contents/Developer"
+    displayName: "Select Xcode $(XCODE_VERSION)"
+  
+  - script: |
+      set -e
+      git clone https://$(GitHubAccessToken)@github.com/xamarinhq/calabash-codesign.git
+      calabash-codesign/apple/create-keychain.sh
+    displayName: "Download and install keychain"
+
+  - script: bundle install
+    displayName: "Bundle install"
+
+  - script: make ipa-agent
+    displayName: "Make ipa-agent"
+
+  - script: make app-agent
+    displayName: "Make app-agent"
+
+  - script: make test-app
+    displayName: "Make test-app"
+
+  - script: make test-ipa
+    displayName: "Make test-ipa"
+
+  - script: make unit-tests
+    displayName: "Make unit-tests"
+
+  - script: |
+      set -e
+      cd cucumber
+      bundle install
+      bundle exec cucumber
+    displayName: "Exec cucumber"
```

Comment:
In order to get an junit xml report, you need to pass flags to cucumber.

In order for this test to make sense, we need to test against the DeviceAgent we just built.

This step needs to call the bin/ci/cucumber.rb script.